crud/crud.cpp

```diff
@@ -1335,46 +1344,28 @@ crud::get_swarm_storage_usage()
 }
 
 
-size_t
-crud::evict_key(const bzn::uuid_t& db_uuid, const bzn::key_t&  key)
+bool
+crud::do_eviction(const database_msg& request, size_t max_size)
 {
-    const auto pair_size = this->storage->get_key_size(db_uuid, key);
-    if (pair_size)
+    const auto PERMS{this->get_database_permissions(request.header().db_uuid()).second};
+    if (auto eviction_policy = this->get_eviction_policy(PERMS))
     {
-        return this->storage->remove(db_uuid, key) == bzn::storage_result::ok ? *pair_size : 0;
-    }
-    return 0;
-}
-
-
-void
-crud::random_cache_replacement(const database_msg& request, size_t key_value_size, size_t max_size, const bzn::key_t& key_exception)
-{
-    const auto [keys, size]{this->storage->get_size(request.header().db_uuid())};
-
-    uint64_t storage_to_free = key_value_size - (max_size - size);
-
-    // We may need to remove one or more key/value pairs to make room for the new one
-    std::hash<std::string> hasher;
-    size_t random_seed = hasher(request.header().request_hash());
-
-    boost::random::mt19937 mt(random_seed);
-    const boost::random::uniform_int_distribution<> dist(0, keys - 1);
-
-    std::vector<bzn::key_t> keys_to_evict;
-
-    const auto available_keys = this->storage->get_keys(request.header().db_uuid());
-    while (storage_to_free)
-    {
-        const auto key_index = dist(mt);
-        const auto key_to_evict = this->storage->get_keys(request.header().db_uuid())[key_index];
-        if ((key_to_evict != key_exception) && (keys_to_evict.end() == std::find(keys_to_evict.begin(), keys_to_evict.end(), key_to_evict)))
+        auto keys_to_evict {eviction_policy->keys_to_evict(request, max_size)};
+        if (keys_to_evict.empty())
         {
-            const size_t evicted_size = evict_key(request.header().db_uuid(), key_to_evict);
-            storage_to_free -= evicted_size < storage_to_free ? evicted_size : storage_to_free;
-            keys_to_evict.emplace_back(key_to_evict);
+            return false;
         }
+
+        std::for_each( keys_to_evict.begin(), keys_to_evict.end()
+                , [&](const auto& key)
```

Comment:
put , on the line above and indent lambda 4 spaces

Reply:
Done